clippy_lints/src/utils/conf.rs

```diff
@@ -238,6 +241,15 @@ pub fn read(path: &Path) -> (Conf, Vec<Error>) {
                 errors.push(Error::Toml(cyc_err));
             }
 
+            let msrv: Option<&String> = toml_ref.msrv.as_ref();
+
+            if msrv.is_some() {
+                match parse_msrv(msrv.cloned(), None, None) {
+                    None => errors.push(Error::Toml(format!("`{}` is not a valid Rust version!", msrv.unwrap()))),
+                    _ => (),
+                }
+            }
+
```

Comment:
Why not save the parsed Version here? This would also prevent having to parse it N times in `lib.rs`.
```suggestion
            let msrv: Option<&String> = toml_ref.msrv.as_ref();

            let msrv = msrv
                .and_then(|s| parse_msrv(msrv, None, None))
                .or_else(|| {
                    errors.push(
                        Error::Toml(
                            format!("`{}` is not a valid Rust version!", msrv.unwrap())
                        )
                    );
                    None
                });
```

Reply:
@flip1995 sorry I didn't quite understand this, could you please elaborate a bit? I'd parsed the msrv multiple times in `lib.rs` as it was being moved into multiple closures

Reply:
You could parse the msrv once and then clone the parsed `Version`, instead of cloning the MSRV `String` N times and then parsing it N times.

Reply:
`parse_msrv` returns `None` if the version is invalid or if the version isn't specified so in the second case calling `msrv.unwrap()` would throw a compiler error. Should `parse_msrv's` behavior be changed?

Reply:
Where would we have to call `msrv.unwrap()`? 

(I also just changed my suggestion above, because I realized, that it was incorrect)

Reply:
Ah do you mean, that I wrote `parse_msrv(msrv, None, None)`? That was a typo and should be `parse_msrv(s, None, None)`.

Reply:
@Suyash458 I have the feeling we're talking about 2 completely different things, so feel free to ask any question. There are no stupid questions :)

Reply:
@flip1995 I was talking about the version of the suggested snippet where the `or_else` call was on the result of `and_then` and not on the result of `parse_msrv`. 

```rust
let msrv = msrv
    .and_then(|s| parse_msrv(s, None, None))
    .or_else(|| {
        errors.push(
            Error::Toml(
                format!("`{}` is not a valid Rust version", msrv.unwrap())
            )
        );
        None
    })
```

I guess this would've thrown an error on `msrv.unwrap()` if it was `None` to begin with